yarn-project/end-to-end/src/e2e_key_registry.test.ts

```diff
@@ -0,0 +1,252 @@
+    let accountAddedToRegistry: AztecAddress;
+
+    it('should generate and register with original keys', async () => {
+      const partialAddress = new Fr(69);
+
+      const masterNullifierPublicKey = new Fr(12);
+      const masterIncomingViewingPublicKey = new Fr(34);
+      const masterOutgoingViewingPublicKey = new Fr(56);
+      const masterTaggingPublicKey = new Fr(78);
+
+      const publicKeysHash = poseidon2Hash([
+        masterNullifierPublicKey,
+        masterIncomingViewingPublicKey,
+        masterOutgoingViewingPublicKey,
+        masterTaggingPublicKey,
+        GeneratorIndex.PUBLIC_KEYS_HASH,
+      ]);
+
+      // We hash the partial address and the public keys hash to get the account address
+      // TODO(#5726): Should GeneratorIndex.CONTRACT_ADDRESS be removed given that we introduced CONTRACT_ADDRESS_V1?
+      // TODO(#5726): Move the following line to AztecAddress class?
+      accountAddedToRegistry = poseidon2Hash([partialAddress, publicKeysHash, GeneratorIndex.CONTRACT_ADDRESS_V1]);
+
+      await keyRegistry
+        .withWallet(wallets[0])
+        .methods.register_from_preimage(
+          AztecAddress.fromField(accountAddedToRegistry),
+          partialAddress,
+          masterNullifierPublicKey,
+          masterIncomingViewingPublicKey,
+          masterOutgoingViewingPublicKey,
+          masterTaggingPublicKey,
+        )
+        .send()
+        .wait();
+    });
+
+    it('checks our registry contract from state vars contract and fails because the address has not been registered yet', async () => {
+      const { txHash } = await stateVarsContract.methods
+        .test_shared_mutable_private_getter_for_registry_contract(keyRegistry.address, 1, accountAddedToRegistry)
+        .send()
+        .wait();
+
+      const rawLogs = await pxe.getUnencryptedLogs({ txHash });
+      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(Fr.ZERO);
+    });
+
+    it('checks our registry contract from state vars contract and finds the address and associated nullifier public key after a delay', async () => {
+      await delay(5);
+
+      const { txHash } = await stateVarsContract.methods
+        .test_shared_mutable_private_getter_for_registry_contract(keyRegistry.address, 1, accountAddedToRegistry)
+        .send()
+        .wait();
+
+      const rawLogs = await pxe.getUnencryptedLogs({ txHash });
+
+      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(new Fr(12));
+    });
+  });
+
+  describe('key rotation flow', () => {
+    it('we rotate the nullifier key', async () => {
+      // This changes
+      const newMasterNullifierPublicKey = new Fr(910);
+
+      await keyRegistry
+        .withWallet(wallets[0])
+        .methods.rotate_keys(wallets[0].getAddress(), newMasterNullifierPublicKey)
+        .send()
+        .wait();
+    });
+
+    it("checks our registry contract from state vars contract and finds our old public key because the key rotation  hasn't been applied yet", async () => {
```

Comment:
```suggestion
    it("checks our registry contract from state vars contract and finds our old public key because the key rotation hasn't been applied yet", async () => {
```